shared/js/background/features/fire-button.js

```diff
@@ -0,0 +1,305 @@
+        // only show the current site option if this an origin we can clear
+        if (currentTabUrl.startsWith('http:') || currentTabUrl.startsWith('https:')) {
+            const origins = getOriginsForUrl(currentTabUrl)
+            const tabsMatchingOrigin = allTabs.filter(tabMatchesHostFilter(origins)).length
+            const pinnedMatchingOrigin = pinnedTabs.filter(tabMatchesHostFilter(origins)).length
+            options.push({
+                name: 'CurrentSite',
+                options: {
+                    origins
+                },
+                descriptionStats: {
+                    ...defaultStats,
+                    openTabs: closeTabs ? tabsMatchingOrigin : 0,
+                    pinnedTabs: closeTabs ? pinnedMatchingOrigin : 0,
+                    cookies: 1,
+                    duration: 'all',
+                    site: getDomain(origins[0], tldtsOptions) || ''
+                }
+            })
+        }
+
+        options.push({
+            name: 'LastHour',
+            options: {
+                since: Date.now() - ONE_HOUR
+            },
+            descriptionStats: {
+                ...defaultStats,
+                duration: 'hour'
+            }
+        })
+        options.push({
+            name: 'Last24Hour',
+            options: {
+                since: Date.now() - (24 * ONE_HOUR)
+            },
+            descriptionStats: {
+                ...defaultStats,
+                duration: 'day'
+            }
+        })
+        options.push({
+            name: 'Last7days',
+            options: {
+                since: Date.now() - (7 * 24 * ONE_HOUR)
+            },
+            descriptionStats: {
+                ...defaultStats,
+                duration: 'week'
+            }
+        })
+        options.push({
+            name: 'Last4Weeks',
+            options: {
+                since: Date.now() - (4 * 7 * 24 * ONE_HOUR)
+            },
+            descriptionStats: {
+                ...defaultStats,
+                duration: 'month'
+            }
+        })
+        options.push({
+            name: 'AllTime',
+            options: {},
+            descriptionStats: {
+                ...defaultStats,
+                duration: 'all'
+            }
+        })
+
+        // mark selected site
+        options.forEach((option) => {
+            if (option.name === selectedOption) {
+                option.selected = true
+            }
+        })
+
+        return {
+            options
+        }
+    }
+
+    async onFireAnimationComplete (msg, sender) {
+        const fireTabId = sender.tab.id
+        chrome.tabs.update(fireTabId, {
+            url: 'chrome://newtab'
+        })
+    }
+}
+
+/**
+ * Given a URL, return a list of origins for that URL, including both secure and insecure contexts,
+ * and using all subdomains up to eTLD+1.
+ * @param {string} url
+ * @returns {string[]} List of origins
+ */
+export function getOriginsForUrl (url) {
```

Comment:
Does the tldts library not have a similar utility?

Reply:
No, though we use their extraction of subdomains as part of this function.